chaoslib/litmus/azure-instance-stop/lib/azure-instance-stop.go

```diff
@@ -211,29 +236,42 @@ func injectChaosInParallelMode(experimentsDetails *experimentTypes.ExperimentDet
+		if experimentsDetails.IsScaleSet == "true" {
+			scaleSetName, vmId := azureStatus.GetScaleSetNameAndInstanceId(vmName)
+			instanceState, err = azureStatus.GetAzureScaleSetInstanceStatus(experimentsDetails.SubscriptionID, experimentsDetails.ResourceGroup, scaleSetName, vmId)
+		} else {
+			instanceState, err = azureStatus.GetAzureInstanceStatus(experimentsDetails.SubscriptionID, experimentsDetails.ResourceGroup, vmName)
+		}
 		if err != nil {
 			log.Errorf("fail to get instance status when an abort signal is received, err: %v", err)
 		}
-		if instanceState != "VM running" {
+		if instanceState != "VM running" && instanceState != "VM starting" {
 			log.Info("[Abort]: Waiting for the Azure instance to get down")
-			if err := azureStatus.WaitForAzureComputeDown(experimentsDetails.Timeout, experimentsDetails.Delay, experimentsDetails.SubscriptionID, experimentsDetails.ResourceGroup, vmName); err != nil {
+			if err := azureStatus.WaitForAzureComputeDown(experimentsDetails.Timeout, experimentsDetails.Delay, experimentsDetails.IsScaleSet, experimentsDetails.SubscriptionID, experimentsDetails.ResourceGroup, vmName); err != nil {
 				log.Errorf("unable to wait till stop of the instance, err: %v", err)
 			}
 
-			log.Info("[Chaos]: Starting back the Azure instance")
-			if err := azureStatus.AzureInstanceStart(experimentsDetails.Timeout, experimentsDetails.Delay, experimentsDetails.SubscriptionID, experimentsDetails.ResourceGroup, vmName); err != nil {
-				log.Errorf("instance power on status check failed, err: %v", err)
-			}
-
 			log.Info("[Abort]: Starting Azure instance as abort signal received")
-			err := azureStatus.WaitForAzureComputeUp(experimentsDetails.Timeout, experimentsDetails.Delay, experimentsDetails.SubscriptionID, experimentsDetails.ResourceGroup, vmName)
-			if err != nil {
-				log.Errorf("Azure instance failed to start when an abort signal is recieved, err: %v", err)
+			if experimentsDetails.IsScaleSet == "true" {
+				if err := azureStatus.AzureScaleSetInstanceStart(experimentsDetails.Timeout, experimentsDetails.Delay, experimentsDetails.SubscriptionID, experimentsDetails.ResourceGroup, vmName); err != nil {
+					log.Errorf("unable to start the Azure instance, err: %v", err)
+				}
+			} else {
+				if err := azureStatus.AzureInstanceStart(experimentsDetails.Timeout, experimentsDetails.Delay, experimentsDetails.SubscriptionID, experimentsDetails.ResourceGroup, vmName); err != nil {
+					log.Errorf("unable to start the Azure instance, err: %v", err)
+				}
 			}
 		}
+
+		log.Info("[Abort]: Waiting for the Azure instance to start")
+		err := azureStatus.WaitForAzureComputeUp(experimentsDetails.Timeout, experimentsDetails.Delay, experimentsDetails.IsScaleSet, experimentsDetails.SubscriptionID, experimentsDetails.ResourceGroup, vmName)
+		if err != nil {
+			log.Errorf("Azure instance failed to start when an abort signal is recieved, err: %v", err)
```

Comment:
```suggestion
			log.Errorf("Azure instance failed to start when an abort signal is recieved, err: %v", err)
```
Azure instance failed to start when an abort signal is received